docs/DeveloperGuide.adoc

```diff
@@ -218,6 +218,42 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+// tag::removeTag[]
+=== removeTag command
+
+The removeTag command removes a tag with the specified name from the address book. All the appearances of the tag in
+the address book are removed. This command is implemented in the style of the `edit` command and `delete` command.
+
+=== Design Considerations
+
+**Aspect:** Implementation for removeTag command +
+**Alternative 1 (current choice):** The command is implemented using methods which remove the tag from each person's tag list. +
+**Pros:** Implementation is modular and less dependent on methods used for other commands. +
+**Cons:** More tedious to implement since it is not making use of the existing resources such as `edit` command. +
+**Alternative 2:** Adapt current `edit` command so that removing the tag is done by editing every person's tag list. +
+**Pros:** Making use of existing resources and easier implementation. +
+**Cons:** More dependency between commands and the relationship between `edit` and `remove` is not clear.
+
+---
+
+**Aspect:** Type of command implemented +
+**Alternative 1 (current choice):** `removeTag` command extends `UndoableCommand` instead of `Command`. +
+**Pros:** Accidental removal of tags can be recovered. +
+**Cons:** Additional memory needed for recording the previous stage of the address book. +
+**Alternative 2:** `removeTag` command extends `Command`. +
+**Pros:** No additional memory needed for recording the previous stage of the address book. +
+**Cons:** As the removal of Tag is from all the persons in address book, an accidental removal can be disastrous if the user does not intend a complete removal.
+
+---
+
+**Aspect:** Removal of the tag from all instead of some by user's choice +
+**Alternative 1 (current choice):** The command currently does not support the removal of a tag from persons specified in the user input.
```

Comment:
Missing plus sign at the end of this line causes previous line to be on the same line as this line...